### DISCUSSION TITLE
Revisiting "no warnings"

### DISCUSSION THREAD

**Discussion by mlutze** [flix/flix#7765] [Ideas]
Perhaps the most common complaint among new users of Flix is the need to `_` every unused variable. This is done in the name of safety, and I'm sure many bugs have been caught or prevented in this way. But it comes at the cost of productivity, as it slows down rapid prototyping of programs, prevents commenting out code easily, etc.

I think we need to visit the "no warnings" principle, as well as the "one language" principle. In particular, it seems best that `RedundancyError`s and their ilk be warnings when building a prototype, but errors when building a finished product.

**Comment by JonathanStarup** [comment]
I agree with the problem - but another way to solve it is to mark things as WIP and have them be allowed to break these things.

Maybe things are automatically WIP (`you haven't fixed your redundancy errors yet, so you cannot call this function`)

**Reply by mlutze** [reply]
But that still means I cannot call a function with some redundancy error in it. If I am rapidly prototyping or debugging, I want to be able to do that.

**Reply by JonathanStarup** [reply]
Right - I guess you could generalize to WIP function, modules, projects

**Reply by JonathanStarup** [reply]
But maybe you are right, actually, why not just have the information available and clearly noticeable - surely anything beyond that cannot decrease errors by much

**Comment by magnus-madsen** [comment]
We can discuss in our next workshop meeting.
